### ISSUE TITLE
backslashes in shell mode on Windows

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#10621]
When you enter a file name in shell mode on Windows, you have to do that awful double-backslash business. Can we avoid having to do this without breaking everything?

**Comment by ihnorton** [comment]
Forward slashes work too.
```
julia> readdir("C:/Windows")
125-element Array{Union(ASCIIString,UTF8String),1}:
 "addins"
 "aksdrvsetup.log"
...
```

**Comment by dhoegh** [comment]
As @ihnorton says forward slashes works to

The auto completion just do it with backslashes. In 0.3 came out with auto completion to forward slashes, which worked perfectly. Then 47d3ecf85e3590c80391dc9b0bbe041f219fbec1 introduced the auto completion with backslashes, which lead to #8483 due to it was not properly escaped. I corrected the escapetion in #8693, by the change on line https://github.com/JuliaLang/julia/pull/8693/files#diff-5533e89f3c15b59d9e1bdffd874daa7cR130. The auto completion could be made to auto complete forward slashes again by changing this line to `push!(matches, id ? file *  "/" ) : file)`. This would require rewriting of some tests but else I do not think there should be any complications.

**Comment by dhoegh** [comment]
Are there any actions to take here, because you can use forward slashes today. We could change the REPLCompletion for paths to finish with forward slash again.
